consensus/core/src/dag_state.rs

```diff
@@ -681,18 +684,44 @@ impl DagState {
     fn eviction_round(commit_round: Round, cached_rounds: Round) -> Round {
         commit_round.saturating_sub(cached_rounds)
     }
+
+    // Leader blocks start from round 1 as we do not consider any blocks from genesis
+    // roudn as a leader block.
+    // TODO: confirm pipelined & multi-leader cases work properly
+    #[cfg(test)]
+    pub(crate) fn get_all_uncommitted_leader_blocks(
```

Comment:
Why not keep this function in test_dag.rs, or move it to test_dag_builder.rs or another location? It seems no DagState internal function is called.

I guess it is easier to implement this way, but it seems this function will not play well with a DagState used "normally". For example if the DagState flushes out some blocks, then iterating from round 1 will not get back any result.